### ISSUE TITLE
mypy doesn't catch error if yield and return are in same function

### ISSUE THREAD

**Issue by justinaustin** [python/mypy#9444]
**Bug Report**

If both `yield` and `return` are in your function, it looks like mypy doesn't pick up issues with incompatible return types.

**To Reproduce**

```python
from typing import Iterable

def foo(x: bool) -> Iterable[int]:
    if x:
        return None
    else:
        yield 1
```

**Expected Behavior**

I expected mypy to return `test.py:5: error: Incompatible return value type (got "None", expected "Iterable[int]")`

**Actual Behavior**

`Success: no issues found in 1 source file`

**Your Environment**

- Mypy version used: 0.782
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.7.7
- Operating system and version: MacOS 10.15.5

**More info**
Interestingly, if I have either
```python
from typing import Iterable

def foo(x: bool) -> Iterable[int]:
    if x:
        yield None
    else:
        yield 1
```
or
```python
def foo(x: bool) -> Iterable[int]:
    if x:
        return None
    else:
        return [1]
```
mypy correctly finds fault with the `yield/return None` block. It seems like mypy only returns Success if both yield and return are in the function together.

**Comment by JelleZijlstra** [comment]
`yield` and `return` do different things; I don't see a bug here. See https://www.python.org/dev/peps/pep-0380/#formal-semantics which introduced support for returning a value from a generator.

**Comment by justinaustin** [comment]
Ah, I did not realize that
```
In a generator, the statement

return value

is semantically equivalent to

raise StopIteration(value)

except that, as currently, the exception cannot be caught by except clauses within the returning generator.
```

Thank you for the quick response!
